include/swift/AST/DiagnosticEngine.h

```diff
@@ -742,8 +743,25 @@ namespace swift {
     void setLocalization(std::string locale, std::string path) {
       assert(!locale.empty());
       assert(!path.empty());
-      localization =
-          std::make_unique<diag::YAMLLocalizationProducer>(locale, path);
+      llvm::SmallString<128> filePath(path);
+      llvm::sys::path::append(filePath, locale);
+      llvm::sys::path::replace_extension(filePath, ".db");
+
+      // If the serialized diagnostics file not available,
+      // fallback to the `YAML` file.
+      if (llvm::sys::fs::exists(filePath)) {
+        auto file = llvm::MemoryBuffer::getFile(filePath);
+        localization =
+            std::make_unique<diag::SerializedLocalizationProducer>(file.get());
```

Comment:
I thought that since we are _sure_ that the file exists (because we already check at L752 `if (llvm::sys::fs::exists(filePath)) `. We don't need to verify that we have a file buffer, because we will always do have a file buffer. WDYT?

Reply:
Existence is one thing, but there could be I/O error while reading or memory mapping it.